### ISSUE TITLE
Can On Demand support something like json_struct?

### ISSUE THREAD

**Issue by lemire** [simdjson/simdjson#1249]
It seems that it would be doable to get On Demand to support something like json_struct.

https://github.com/jorgen/json_struct

**Comment by jkeiser** [comment]
It definitely seems likely. Ultimately I'd like to see an interchangeable api where you can use json path to find objects, declarative deserialization to get individual records, and drop back down to ondemand for anything more complicated. ondemand is likely the low level language for a couple of nice higher level capabilities.

**Comment by jkeiser** [comment]
This is also a nice way to make efficient order insensitive object deserialization. It can even use hashes when it finds the object out of order.

**Comment by jkeiser** [comment]
Related: #1382.

**Comment by jkeiser** [comment]
What's your read on having this for 1.0 now? It will likely take some ingenuity to do right ...

**Comment by lemire** [comment]
We can delay it post 1.0.

**Comment by lemire** [comment]
Yes, and it does now. On Demand supports json_struct-style deserialization in several ways:
- custom types through `tag_invoke` (C++20); see "Use `tag_invoke` for custom types" in doc/basics.md;
- fully automatic `doc.get<MyStruct>()` with C++26 static reflection (#2282);
- selective extraction into a struct with `extract_into<"make", "model">(car)`;
- single-pass, order-independent field extraction with key selectors (#2776, on `master`, shipping in the next release).

Closing as implemented.
